### ISSUE TITLE
[a11y]: aria-controls properties controlled by Data tables are not valid

### ISSUE THREAD

**Issue by patrickobrien1** [carbon-design-system/carbon#20616]
### Package

@carbon/react

### Browser

Firefox

### Operating System

MacOS

### Package version

1.76.0

### React version

16.14.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Assistive technology

VoiceOver

### Description

<img width="1911" height="1034" alt="Image" src="https://github.com/user-attachments/assets/cce7138d-2f5a-4b4a-b292-17d934963a01" />
'id' "expanded-row-96d1ebc0-dc1a-4b96-9f04-06709c059301" specified for the ARIA property 'aria-controls' value is not valid. Same issue applicable to all the expand row buttons in the table

Both the aria-controls and id props that are created should utilize the getTablePrefix() function in the DataTable.tsx file so that multiple data tables on the same page do not conflict, and ids match exactly.

### WCAG 2.1 Violation

WCAG 4.1.2 Name, Role, Value

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-datatable-basic--overview

### Steps to reproduce

Utilize any data table that contains expandable rows.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
